doc/reactions.md

```diff
@@ -0,0 +1,152 @@
+[<- Back to Index](index.md)
+
+# Reactions
+
+Reactions allow SQL LRS to watch for patterns in submitted xAPI data and dynamically generate new statements in response.
+
+## Usage
+
+To use Reactions the `LRSQL_ENABLE_REACTIONS` environment variable or the `enableReactions` LRS configuration property must be set to `true`. Reactions are disabled by default.
+
+Reactions are defined in JSON:
+
+``` json
+
+{
+  "identityPaths": [
+    [
+      "actor",
+      "mbox"
+    ]
+  ],
+  "conditions": {
+    "a": {
+      "and": [
+        {
+          "path": [
+            "object",
+            "id"
+          ],
+          "op": "eq",
+          "val": "https://example.com/activities/a"
+        },
+        {
+          "path": [
+            "verb",
+            "id"
+          ],
+          "op": "eq",
+          "val": "https://example.com/verbs/completed"
+        },
+        {
+          "path": [
+            "result",
+            "success"
+          ],
+          "op": "eq",
+          "val": true
+        }
+      ]
+    },
+    "b": {
+      "and": [
+        {
+          "path": [
+            "object",
+            "id"
+          ],
+          "op": "eq",
+          "val": "https://example.com/activities/b"
+        },
+        {
+          "path": [
+            "verb",
+            "id"
+          ],
+          "op": "eq",
+          "val": "https://example.com/verbs/completed"
+        },
+        {
+          "path": [
+            "result",
+            "success"
+          ],
+          "op": "eq",
+          "val": true
+        },
+        {
+          "path": [
+            "timestamp"
+          ],
+          "op": "gt",
+          "ref": {
+            "condition": "a",
+            "path": [
+              "timestamp"
+            ]
+          }
+        }
+      ]
+    }
+  },
+  "template": {
+    "actor": {
+      "mbox": {
+        "$templatePath": [
+          "a",
+          "actor",
+          "mbox"
+        ]
+      }
+    },
+    "verb": {
+      "id": "https://example.com/verbs/completed"
+    },
+    "object": {
+      "id": "https://example.com/activities/a-and-b",
+      "objectType": "Activity"
+    }
+  }
+}
+
+
+```
+
+### Identity Paths
+
+`identityPaths` is an array of zero or more paths that must match across all statements in the sequence. These could include an actor identifier or statement registration.
```

Comment:
In this above sentence, I'd add the following: "In the above reaction, the `["actor", "mbox"]` path points to the `mbox` value in every statement being inserted into the LRS. Every statement therefore must contain an `mbox` as its actor IFI.*"

*I'm actually not sure what happens if you insert a non-matching statement, so that should also be added to this section.

Reply:
I actually wrote a tooltip that does a dive on this. You can crib from it if its useful.

`"USE WITH CAUTION. Identity Paths are a method of grouping statements for which you are attempting to match conditions. Typically, Reactions may revolve around actor, e.g. `$.actor.mbox` or `$.actor.account.name` which is equivalent to saying \"For a given Actor, look for statements that match the Conditions above\". This is what the default is set to. Alternative approaches to Identity Path may be used by modifying this section, for instance `$.context.registration` to group statements by learning session."`

Reply:
ad895b213934129312a4b0bd8db040d6b5733a72